ert_shared/ensemble_evaluator/entity/ensemble.py

```diff
@@ -7,60 +12,124 @@
 )
 from res.enkf import EnKFState
 
+from pathlib import Path
+from collections import defaultdict
+from graphlib import TopologicalSorter
+
 
 logger = logging.getLogger(__name__)
 
 
+def _sort_steps(steps: typing.List["_Step"]):
+    graph = defaultdict(set)
+    if len(steps) == 1:
+        return (steps[0].get_name(),)
+    edged_nodes = set()
```

Comment:
What is the purpose of the `edge_nodes`? They don't seem to be used?

Reply:
The `edged_nodes` is used to subtract from all nodes (steps) such that you're left with nodes without edges (`null_nodes`). These are then added to the graph.

Reply:
Aha, I get it now! `[graph[node] for node in null_nodes]` suddenly makes sense. Could we make it a bit move explicit? Perhaps something like:

```python
for node in null_nodes:
    graph[node] = {}
```

Also, the most standard term for such vertices is `isolated_nodes`

Reply:
🤓

Reply:
😎